crates/bevy_ui/src/measurement.rs

```diff
@@ -0,0 +1,77 @@
+use bevy_ecs::prelude::Component;
+use bevy_math::Vec2;
+use bevy_reflect::Reflect;
+use std::fmt::Formatter;
+pub use taffy::style::AvailableSpace;
+
+impl std::fmt::Debug for CalculatedSize {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("CalculatedSize").finish()
+    }
+}
+
+/// A `Measure` is used to compute the size of a ui node
+/// when the size of that node is based on its content.
+pub trait Measure: Send + Sync + 'static {
+    /// Calculate the size of the node given the constraints.
+    fn measure(
+        &self,
+        width: Option<f32>,
+        height: Option<f32>,
+        available_width: AvailableSpace,
+        available_height: AvailableSpace,
+    ) -> Vec2;
+
+    /// Clone and box self.
+    fn dyn_clone(&self) -> Box<dyn Measure>;
+}
+
+/// A `FixedMeasure` is a `Measure` that ignores all constraints and
+/// always returns the same size.
+#[derive(Default, Clone)]
+pub struct FixedMeasure {
+    size: Vec2,
+}
+
+impl Measure for FixedMeasure {
+    fn measure(
+        &self,
+        _: Option<f32>,
+        _: Option<f32>,
+        _: AvailableSpace,
+        _: AvailableSpace,
+    ) -> Vec2 {
+        self.size
+    }
+
+    fn dyn_clone(&self) -> Box<dyn Measure> {
+        Box::new(self.clone())
+    }
```

Comment:
could this be moved to a default impl on the trait?

Reply:
It doesn't seem to be possible because of the object-safe rules.

Reply:
I've had an idea for an alternative implementation that wouldn't need the dyn_clone function at all. I'll try it when I get home in a few hours.

Reply:
Can be done but it's a bigger problem than I thought. It would need a lot of changes and bug fixes outside of the scope of this PR.